pylint/lint/expand_modules.py

```diff
@@ -131,11 +128,14 @@ def expand_modules(
                     "basepath": filepath,
                     "basename": modname,
                 }
-        has_init = (
-            not (modname.endswith(".__init__") or modname == "__init__")
-            and os.path.basename(filepath) == "__init__.py"
-        )
-        if has_init or is_namespace or is_directory:
+        if (
+            is_namespace
+            or is_directory
+            or (
+                not (modname.endswith(".__init__") or modname == "__init__")
+                and os.path.basename(filepath) == "__init__.py"
+            )
+        ):
```

Comment:
This strikes me as an instance of something becoming worse.

Reply:
If I'm reading it right, this is actually a micro performance improvement. `is_namespace` and `is_directory` are already sitting around, so calculating `has_init` before checking them is potentially unnecessary.

I guess the name `has_init` gives a little clue. I could add a comment to that effect if you like.

Reply:
Nah, no need to touch this specific example, was just saying that some of these locals seemed to be here for self-documenting. (Although you have a point that there's a micro-optimization to be had.) I suppose my thought about doing this PR was why remove nicely self-documenting locals just to get the locals count down? Won't stand in the way though if my view is not seconded 🗣️

Reply:
Yes, there is a risk of sacrificing readability just for the sake of chasing stats. But my feeling from looking at these functions is that they have reached a point where the marginal value of additional variables is negative. It can be bewildering to keep track of what all these variables mean, particularly when so many are used only once.

I'm happy to add some back though, or add comments.